x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java

```diff
@@ -85,6 +95,28 @@ public void testInbound() throws Exception {
         verify(authzService).authorize(eq(authentication), eq("_action"), eq(request), anyActionListener());
     }
 
+    public void testRemoteAccessInbound() {
+        TransportRequest request = mock(TransportRequest.class);
+        Authentication authentication = AuthenticationTestHelper.builder().internal(SystemUser.INSTANCE).build();
+        boolean allowlisted = randomBoolean();
+        String action = allowlisted ? randomFrom(SecurityServerTransportInterceptor.REMOTE_ACCESS_ACTION_ALLOWLIST) : "_action";
+        doAnswer(getAnswer(authentication)).when(authcService).authenticate(eq(action), eq(request), eq(true), anyActionListener());
+        doAnswer(getAnswer(authentication)).when(remoteAccessAuthcService)
+            .authenticate(eq(action), eq(request), eq(false), anyActionListener());
+        ServerTransportFilter filter = getNodeRemoteAccessFilter();
+        PlainActionFuture<Void> future = new PlainActionFuture<>();
+        filter.inbound(action, request, channel, future);
+        // future.get(); // don't block it's not called really just mocked
+        verify(authzService).authorize(eq(authentication), eq(action), eq(request), anyActionListener());
+        if (allowlisted) {
+            verify(remoteAccessAuthcService).authenticate(anyString(), any(), anyBoolean(), anyActionListener());
+            verify(authcService, never()).authenticate(anyString(), any(), anyBoolean(), anyActionListener());
+        } else {
+            verify(authcService).authenticate(anyString(), any(), anyBoolean(), anyActionListener());
```

Comment:
This will change once we've updated allowlist behavior on FC & QC side.

Reply:
Can we add a TODO?